pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -119,107 +117,139 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// Add conditions if there are none
+	if instance.Status.Conditions == nil {
+		initConditions(&instance.Status.Conditions)
+
+		err = r.client.Status().Update(context.TODO(), instance)
+		if err != nil {
+			reqLogger.Error(err, "Failed to add conditions to status")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Create happy conditions
+	initHappyConditions(&r.conditions)
```

Comment:
Our state isn't always happy here.  If the previous Reconcile was a Failure, why would we set it to Happy before we have fixed the issue?

Reply:
I know that I'm not using the `initHappyConditions()` anymore but the concept is still the same. We have an in-memory representation `r.conditions` and the persistent representation `instance.Status.Conditions`. What you'll see in my latest commit is that I always set `r.conditions = nil` and as we go through the different `ensureXYZ` they will set conditions on `r.conditions` (notably not `instance.Status.Conditions`) and the last step of `Reconcile()` is to  handle the case where `r.conditions != nil` (means something like KubeVirt is not available) or `r.conditions == nil` (means that everyone is in a happy state).